src/distributed-command.js

```diff
@@ -47,43 +44,42 @@ class TerraformCommand extends AbstractCommand {
       .addOption('git-diff', 'g', 'List of components to include (git diff)', Array, [])
       .addOption('var', 'r', 'Variable(s) to be used by terraform', Array, [])
       .addOption('var-file', 'l', 'Variable file(s) to be used by terraform', Array, [])
-      .addOption('dependency', 'p', 'Set dependency validation strategy (auto, ignore, include)', String, 'auto')
-    ;
+      .addOption('dependency', 'p', 'Set dependency validation strategy (auto, ignore, include)', String, 'auto');
   }
 
   /**
+   *
    * @returns {Promise}
    */
-  validate() {
-    return super.validate().then(token => {
-      this._tokenIsValid = token;
+  async validate() {
+    try {
+      this._tokenIsValid = await super.validate();
 
-      if (this._tokenIsValid && logs) {
+      if (this._tokenIsValid && this.terrahubCfg.logs) {
         this.logger.updateContext({ canLogBeSentToApi: true });
       }
 
-      return Promise.resolve();
-    }).then(() => this._checkProjectDataMissing()).then(() => {
-      if (this._isComponentsCountZero() && this.getName() !== 'configure') {
+      await this.checkProjectDataMissing();
+
+      if (this.isComponentsCountZero() && this.getName() !== 'configure') {
         throw new Error('No components defined in configuration file. '
           + 'Please create new component or include existing one with `terrahub component`');
       }
 
-      return Promise.resolve();
-    }).then(() => {
       const nonExistingComponents = this._getNonExistingComponents();
 
       if (nonExistingComponents.length) {
         throw new Error('Some of components were not found: ' + nonExistingComponents.join(', '));
       }
 
-      return Promise.resolve();
-    }).then(() => this._checkProjectDuplicateComponents())
-      .catch(err => {
-        const error = err.constructor === String ? new Error(err) : err;
+      const fullConfig = this.getExtendedConfig();
+      return this.checkProjectDuplicateComponents(fullConfig);
```

Comment:
(1) move this feature into the functionality when terrahub components are loaded, that's supposed to be the moment when they must be checked for duplicates.
(2) don't check for duplicates every time; instead, use `~/.terrahub/cache/` to validate once and check only when the list of components changed